be/src/exec/pipeline/sink/file_sink_operator.cpp

```diff
@@ -0,0 +1,300 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+#include "exec/pipeline/sink/file_sink_operator.h"
+
+DIAGNOSTIC_PUSH
+DIAGNOSTIC_IGNORE("-Wclass-memaccess")
+#include "bthread/execution_queue.h"
+DIAGNOSTIC_POP
+#include "column/chunk.h"
+#include "exec/workgroup/scan_executor.h"
+#include "exec/workgroup/scan_task_queue.h"
+#include "exprs/expr.h"
+#include "runtime/buffer_control_block.h"
+#include "runtime/query_statistics.h"
+#include "runtime/result_buffer_mgr.h"
+#include "runtime/runtime_state.h"
+#include "udf/java/utils.h"
+#include "util/defer_op.h"
+#include "util/spinlock.h"
+
+namespace starrocks::pipeline {
+
+// FileSinkBuffer accepts input from all FileSinkOperators, it uses an execution queue to asynchronously write chunks to file one by one.
+class FileSinkBuffer {
+public:
+    FileSinkBuffer(std::vector<ExprContext*>& output_expr_ctxs, std::shared_ptr<ResultFileOptions> file_opts,
```

Comment:
do we have the buffer limit and back pressure mechanism ？

Reply:
> do we have the buffer limit and back pressure mechanism ？

Yes, we allow up to 64(kExecutionQueueSizeLimit) pending chunks in execution queue, if this number is exceeded, need_input will return false